contracts/core/Registry.sol

```diff
@@ -1,47 +1,308 @@
 // SPDX-License-Identifier: MIT
-pragma solidity ^0.8.0;
+pragma solidity 0.8.19;
 
-// use Solady Roles for ownership of projects: https://github.com/Vectorized/solady/blob/main/src/auth/OwnableRoles.sol
-contract Registry {
-    struct IdentityDetails {
+import {AccessControl} from "../../lib/openzeppelin-contracts/contracts/access/AccessControl.sol";
+
+import {Metadata} from "./libraries/Metadata.sol";
+
+contract Registry is AccessControl {
```

Comment:
Question whether you want to grant some Allo multisig the `DEFAULT_ADMIN_ROLE` so that it can grant or revoke roles in emergencies.

Reply:
We could have that set via the constructor ?

Reply:
Decided not to implement this